### PR TITLE
Avoid including `yvals.h` when the compiler is not MSVC.

### PR DESCRIPTION
**Pull request by wmaxey**
## Description

closes nvbug/4902968

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: libcudacxx/include/cuda/std/__cccl/assert.h [hunk 1]
```diff
@@ -62,7 +62,7 @@
 //! _CCCL_ASSERT_IMPL_HOST should never be used directly
 #if defined(_CCCL_COMPILER_NVRTC) // There is no host standard library in nvrtc
 #  define _CCCL_ASSERT_IMPL_HOST(expression, message) ((void) 0)
-#elif __has_include(<yvals_core.h>) // MSVC uses _STL_VERIFY from <yvals.h>
+#elif __has_include(<yvals_core.h>) && defined(_CCCL_COMPILER_MSVC) // MSVC uses _STL_VERIFY from <yvals.h>
 #  include <yvals.h>
```
Comment (dkolsen-pgi) [inline libcudacxx/include/cuda/std/__cccl/assert.h:66]: The `__has_include` is checking for a different header file, `<yvals_core.h>`, than the header that is actually included, `<yvals.h.>`.  That looks like a bug that should be fixed, even if it wasn't what triggered the NVBug.

Comment (ericniebler) [reply libcudacxx/include/cuda/std/__cccl/assert.h:66]: +1

Comment (miscco) [reply libcudacxx/include/cuda/std/__cccl/assert.h:66]: Yeah that was me changing the include

Comment (wmaxey) [reply libcudacxx/include/cuda/std/__cccl/assert.h:66]: STL has both. Headers, clang has only the one, @miscco do we need to check for yvals_core.h for Clang's benefit?

Comment (miscco) [reply libcudacxx/include/cuda/std/__cccl/assert.h:66]: No we should only check for the proper MSVC header and I dont mind also checking for MSVC

**Review by griwes — APPROVED** [review]

**Pull request merged**